### PR TITLE
make Panel a widget

### LINKED ISSUE #371: make Panel a widget
right now it just has some util functions for generating titlebars, but it should take the panel contents as an argument

### PR DESCRIPTION
**Pull request by PPER**
#371

### CODE REVIEW

File: src/hazelweb/gui/Panel.re [hunk 1]
```diff
@@ -10,3 +10,29 @@ let view_of_main_title_bar = (title_text: string) =>
...
+      ~id: option(string),
+      ~classes: List.t(string),
+      ~body_contents: List.t(Vdom.Node.t),
+    )
+    : Vdom.Node.t => {
+  switch (id) {
+  | Some(id) =>
+    Vdom.(
+      Node.div(
+        [Attr.id(id), Attr.classes(["panel"] @ classes)],
```
Comment (dm0n3y) [inline src/hazelweb/gui/Panel.re:26]: Cons `"panel"` instead of appending

File: src/hazelweb/gui/Panel.re [hunk 2]
```diff
@@ -10,3 +10,29 @@ let view_of_main_title_bar = (title_text: string) =>
...
+        [view_of_main_title_bar(title_text), ...body_contents],
+      )
+    )
+  | None =>
+    Vdom.(
+      Node.div(
+        [Attr.classes(["panel"] @ classes)],
```
Comment (dm0n3y) [inline src/hazelweb/gui/Panel.re:33]: same here

**Review by dm0n3y — CHANGES_REQUESTED** [review]
Cursor inspector should also use `Panel`, looks good otherwise

**Comment by cyrus-** [comment]
@PPER Marking as draft pending small fixes @dmoon1221 identified above.

**Comment by cyrus-** [comment]
@disconcision @hannahpotter is this useful for you as you add / mess with the sidebar architecture at all? if not we can close.

**Comment by cyrus-** [comment]
haven't heard back, closing

**Pull request closed without merging**